astropy/io/votable/tree.py

```diff
@@ -2658,13 +2674,12 @@ def _resize_strategy(self, size):
         return int(np.ceil(size * RESIZE_AMOUNT))
 
     def add_field(self, field: Field) -> None:
-        self.fields.append(field)
-        self.all_fields.append(field)
```

Comment:
I found that this line was not needed and in fact caused some bugs that are revealed by existing tests once we start using `add_field` instead of direct mutations. This is a regression from #15959, which can easily be fixed independently in a -1/+0 separate PR if desired.